### PR TITLE
eviction manager: add test case with localStorageCapacityIsolation open

### LINKED ISSUE #120062: eviction_manager _test.go should have unit tests that cover localStorageCapacityIsolation
As I was reading the test code, I noticed that we are missing a pretty large chunk of logic around localStorageCapacityIsolation.  

This feature was GA in 1.25 so it would be ideal to have unit test coverage around this code.  

```
	// evict pods if there is a resource usage violation from local volume temporary storage
	// If eviction happens in localStorageEviction function, skip the rest of eviction action
	if m.localStorageCapacityIsolation {
		if evictedPods := m.localStorageEviction(activePods, statsFunc); len(evictedPods) > 0 {
			return evictedPods
		}
	}
```

This code monitors whether or not there is a violation in temporary storage and if there is it would evict the pod.  

One should be able to intiailize the manager with `localStorageCapacityIsolation` of true and then write some test cases to verify this behavior.  

/help
/sig node
/good-first-issue
/kind bug

### PR DESCRIPTION
**Pull request by charles-chenzz**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
this PR add test case for the missing part of manager with localStorageCapacityIsolation open and improve test coverage
#### Which issue(s) this PR fixes:

Fixes #120062 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by charles-chenzz** [comment]
test coverage
before:
image

after:
image

**Review by kannon92 — COMMENTED** [review]
/cc

File: pkg/kubelet/eviction/eviction_manager_test.go [hunk 1]
```diff
@@ -1784,3 +1810,62 @@ func TestUpdateMemcgThreshold(t *testing.T) {
...
+		nodeRef:                       nodeRef,
+		localStorageCapacityIsolation: true,
+		dedicatedImageFs:              &diskInfoProvider.dedicatedImageFs,
+	}
+
+	activePodsFunc := func() []*v1.Pod {
+		return pods
+	}
+
+	mgr.synchronize(diskInfoProvider, activePodsFunc)
```
Comment (kannon92) [inline pkg/kubelet/eviction/eviction_manager_test.go:1870]: You should test the outcome of this function after `synchronize`.

File: pkg/kubelet/eviction/eviction_manager_test.go [hunk 2]
```diff
@@ -117,6 +117,32 @@ func makePodWithDiskStats(name string, priority int32, requests v1.ResourceList,
 	return pod, podStats
 }
 
+func makePodWithLocalStorageCapacityIsolationOpen(name string, priority int32, requests v1.ResourceList, limits v1.ResourceList, memoryWorkingSet string) (*v1.Pod, statsapi.PodStats) {
```
Comment (kannon92) [inline pkg/kubelet/eviction/eviction_manager_test.go:120]: I would rather use an enum for name.

File: pkg/kubelet/eviction/eviction_manager_test.go [hunk 3]
```diff
@@ -1784,3 +1810,62 @@ func TestUpdateMemcgThreshold(t *testing.T) {
 	fakeClock.Step(2 * notifierRefreshInterval)
 	manager.synchronize(diskInfoProvider, activePodsFunc)
 }
+
+func TestManagerWithLocalStorageCapacityIsolationOpen(t *testing.T) {
+	podMaker := makePodWithLocalStorageCapacityIsolationOpen
+	summaryStatsMaker := makeDiskStats
+	podsToMake := []podToMake{
```
Comment (kannon92) [inline pkg/kubelet/eviction/eviction_manager_test.go:1817]: You are missing the default case.

File: pkg/kubelet/eviction/eviction_manager_test.go (hunk 2 above)
Comment (charles-chenzz) [reply pkg/kubelet/eviction/eviction_manager_test.go:120]: I follow the pattern from https://github.com/kubernetes/kubernetes/blob/1494839b32aac4104e105bf02bfc6e477a798f06/pkg/kubelet/eviction/eviction_manager_test.go#L104

and
https://github.com/kubernetes/kubernetes/blob/1494839b32aac4104e105bf02bfc6e477a798f06/pkg/kubelet/eviction/eviction_manager_test.go#L112

File: pkg/kubelet/eviction/eviction_manager_test.go (hunk 1 above)
Comment (charles-chenzz) [reply pkg/kubelet/eviction/eviction_manager_test.go:1870]: the outcome here is a list of pods, are you suggesting using ``reflect.DeepEqual()`` to compare if we evicted the right pod?

Comment (kannon92) [reply pkg/kubelet/eviction/eviction_manager_test.go:1870]: Yea.  Make sure you test that it evicted the right pod.

Comment (kannon92) [reply pkg/kubelet/eviction/eviction_manager_test.go:1870]: And Pod was evicted etc

File: pkg/kubelet/eviction/eviction_manager_test.go (hunk 2 above)
Comment (charles-chenzz) [reply pkg/kubelet/eviction/eviction_manager_test.go:120]: in my case I just use the eviction case name as pod name to match different podStats. So it seems like force code paths.

Comment (charles-chenzz) [reply pkg/kubelet/eviction/eviction_manager_test.go:120]: we can use enum as well, just need to add a field on struct ``podToMake``

Comment (kannon92) [reply pkg/kubelet/eviction/eviction_manager_test.go:120]: I see.  Maybe we can leave it for now and see what a sig-node reviewer says.

File: pkg/kubelet/eviction/eviction_manager_test.go (hunk 1 above)
Comment (kannon92) [reply pkg/kubelet/eviction/eviction_manager_test.go:1870]: Please update this when you get a chance.  Maybe you can follow https://github.com/kubernetes/kubernetes/blob/a0873d37c58757ac6c2a339af1e4f8d601cdef20/pkg/kubelet/eviction/eviction_manager_test.go#L289 as an example for this.

Comment (charles-chenzz) [reply pkg/kubelet/eviction/eviction_manager_test.go]: already updated. PTAL when you get a chance.

**Review by mrunalp — APPROVED** [review]

**Pull request merged**